chapters/projectManagement.typ

```diff
@@ -0,0 +1,208 @@
+= Project Management
+// [] Vorgehen (Prozess, Reviews, Workflows, Qualitätssicherung)
+// [x] Projektplan, 
+//    [] Vergleich ursprüngliche Planung, effektive Ausführung
+// [x] Zeiterfassung (Stunden pro Woche/Stunden pro Task-Kategorie, wie Implementation Doku, Meeting, etc.)
+// Hinweis: Keine Liniendiagramme für die Darstellung von Zeit/Arbeitsaufwand pro Woche
+
+This section describes the approach of the project as well as the project plan and time tracking.
+In @approach, it is explained how the project was approached.
+@project-plan, looks into the planning of the project, including a comparative analysis between the initial plan and its effective implementation.
+Additionally, @time-tracking, provides a summarized overview of the allocated working hours coupled with a reflective assessment of the time invested in the reflects on the time invested.
+
+== Approach <approach>
+
+Development work occurred on both Linux and Windows utilizing CLion with the Ninja build system.
+The language server features were tested using Visual Studio Code and the clangd extension.
+
+To discuss the ongoing process a weekly meeting with the advisor was held were it was discussed what work has been done and what is planned for the next week.
+Receiving a feedback of the ongoing development is quite important as missing work can be detected and adjusted just the week after.
+
+The project is split into three main steps.
++ Project Setup
++ Analysis
++ Implementation and Finalization
+
+/ Project Setup : #[
+  Before implementing a project setup needs to be in place, therefore some research needed to be done to figure out what is needed to extend the LLVM project with new features.
+  The setup needed to work on Linux and Windows as both systems were used for this project.
+]
+
+/ Analysis : #[
+  The analysis part consists of a lot of research on how the clangd language server works and how it is communicating with the IDE.
+  This step also contains gaining all knowledge needed for implementation, such as looking into the idea of concepts and figuring out what refactoring features would be a good addition to have within the language server.
+]
+
+/ Implementation and Finalization: #[
+  In the implementation phase the actual refactoring features are implemented according to the analysis.
+  To make the implementation ready for contribution it needs to be refined, which means the code need to be readable and following the development guidelines @llvm_coding_standards.
+  When the refinement is done a pull request can be created to finalize the implementation step.
+]
+
+#set page(flipped: true)
+
+== Project Plan <project-plan>
+
+The project plan was developed in week 3 after the ideas for the refactoring features were set.
+Without concrete ideas a good project plan can not be worked out.
+Each phase of a refactoring implementation is planned with the same amount of time as the implementation itself.
```

Comment:
Ned sicher was meinsch.
Was isch de Unterschied vo "phase of a refactoring implementation" und "implementation itself"?

Reply:
han die ganz umsetzig gmeint, hans ez "refactoring feature" anstatt "refactoring implementation" drin